### ISSUE TITLE
[ElasticSearch] SearchTerm is ignored when adding a predefined filter to the search-ui config

### ISSUE THREAD

**Issue by ZakKa89** [elastic/search-ui#892]
**Describe the bug**
SearchTerm is ignored when adding a predefined filter to the search-UI config. So doesn't matter what you search on, the search will just return the items from the predefined filters. 

**To Reproduce**

1. add filter to the searchQuery object for example"
```
searchQuery: {
    filters: [{ field: "id", type: "all", values: ["park_acadia"]}],
```

https://codesandbox.io/s/github/elastic/search-ui/tree/master/examples/sandbox?file=/src/pages/elasticsearch/index.js:1108-1121

Note that for App Search this works fine! 

2. search for a value that you know will not give any results like `///\\/\/\/\/\/\/\//\\\`

You will see that you will still get the result of the filter... 

**Expected behavior**
- searchTerm must still work when having predefined filters

**Which backends and packages are you using:**
Backend: [Elastic Search cloud]

    "@elastic/react-search-ui": "^1.17.1",
    "@elastic/react-search-ui-views": "^1.17.1",
    "@elastic/search-ui": "^1.17.1",
    "@elastic/search-ui-elasticsearch-connector": "^1.17.1",

**Comment by JasonStoltz** [comment]
Good catch, I appreciate the bug report.

**Comment by RobbyUitbeijerse** [comment]
hi there! Not sure if related, but we came to an interesting situation and it feels like the right issue to post this in.

The filter we are applying looks like this:

`
      filters: [
        {
          field: 'owner.keyword',
          values: [userAddress],
          type: 'all' as const,
        },
      ],
`

Within the input that you can see in the video below, we would would like to allow users to further specify the results they are seeing by allowing them to set a search term on top of the pre-defined filters in the config. Just to clarify, this is the handler for setting the search term:

```
        onChange={e => setSearchTerm(e.target.value)}
```


Combining this, we do actually see that setSearchTerm seems to be working, but it seems to _sort_ the results rather than filtering out results that can be considered irrelevant by setting the searchTerm. Please note that the items you see in the video is the subset of our total index based on the pre-defined filters. 

https://user-images.githubusercontent.com/10176709/199459008-3575968c-4721-47c0-824a-4fbe238f8b82.mp4

**Comment by ZakKa89** [comment]
> we would would like to allow users to further specify the results they are seeing by allowing them to set a search term on top of the pre-defined filters in the config


@RobbyUitbeijerse Your question is probably not related to my issue, but there's a difference between global filters (predefined) and filters for the user in the search UI config. The filters for the user are defined in the 'facets' object. I don't think you're using that. So I suggest to dive into the Facets configuration part of search-ui

TLDR: You will need to add the filters that the user has control of to the config of Search-UI within the facets object

**Comment by Mites88** [comment]
> hi there! Not sure if related, but we came to an interesting situation and it feels like the right issue to post this in.
> 
> The filter we are applying looks like this:
> 
> ```
>       filters: [
>         {
>           field: 'owner.keyword',
>           values: [userAddress],
>           type: 'all' as const,
>         },
>       ],
> ```
> 
> Within the input that you can see in the video below, we would would like to allow users to further specify the results they are seeing by allowing them to set a search term on top of the pre-defined filters in the config. Just to clarify, this is the handler for setting the search term:
> 
> ```
>         onChange={e => setSearchTerm(e.target.value)}
> ```
> 
> Combining this, we do actually see that setSearchTerm seems to be working, but it seems to _sort_ the results rather than filtering out results that can be considered irrelevant by setting the searchTerm. Please note that the items you see in the video is the subset of our total index based on the pre-defined filters.
> 
>  setSearchTerm.mp4 
> I've tried to debug the query that gets sent over to Elastic by logging the request body. The interesting part is that setting the search term (and filters) client side through the input onChange leads to quite a bit of a different query compared to the an initial query where filters + a search term is applied

My team is facing this issue as well. Have you found a solution or workaround for this?

**Comment by Mites88** [comment]
Elasticsearch connector's `requestBody` query uses this `should` property which seems to cause this behaviour.
![image](https://user-images.githubusercontent.com/22935809/205381814-36fcc3fb-2611-4422-aec1-eb49555a54d9.png)
If you force change it to `must` through [customizing ES connector](https://docs.elastic.co/search-ui/api/connectors/elasticsearch#customise-the-elasticsearch-request-body), the issue seems to be resolved.

**Comment by Scientist-Ansh** [comment]
> @ZakKa89 If you want to help us fix any of this... we are hiring :)

Are you guys still hiring I so want to fix this bug and would love to work on the search-ui 💯

The searchTerm is still ignored if there is a global filter present and the only solution that works is mentioned by @Mites-G
